c7n/resources/cw.py

```diff
@@ -315,3 +316,72 @@ def process_resource_set(self, client, accounts, resources):
             if found:
                 results.append(r)
         return results
+
+
+@LogGroup.action_registry.register('set-encryption')
+class EncryptLogGroup(BaseAction):
+    """Encrypt/Decrypt a log group
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: encrypt-log-group
+            resource: log-group
+            filters:
+              - kmsKeyId: absent
+            actions:
+              - type: set-encryption
+                kms_key_id: kms:key:arn
+                state: True
+
+          - name: decrypt-log-group
+            resource: log-group
+            filters:
+              - kmsKeyId: kms:key:arn
+            actions:
+              - type: set-encryption
+                state: False
+    """
+    schema = type_schema(
+        'set-encryption',
+        kms_key_id={'type': 'string'},
+        kms_key_alias={'type': 'string'},
+        state={'type': 'boolean'})
+    permissions = ('logs:AssociateKmsKey', 'logs:DisassociateKmsKey')
+
+    def validate(self):
+        self.key_id = self.data.get('kms_key_id')
+        self.key_alias = self.data.get('kms_key_alias')
+
+        if not self.data.get('state', True):
+            return self
+
+        if (not self.key_id and not self.key_alias) or \
+                (self.key_id and self.key_alias):
+            raise ValueError('Must specify either a KMS key ARN or Alias')
+        return self
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('logs')
+
+        if self.key_alias:
+            # Get Key ARN
+            alias = 'alias/' + self.key_alias
```

Comment:
this should be conditionally prefixed if the alias is already extant on the configured value.